lib/active_support/cache/database_store/model.rb

```diff
@@ -15,7 +15,13 @@ def self.truncate!
         scope :created_before, ->(date = 1.month.ago) { where(arel_table[:created_at].lt(date)) }
 
         def self.namespaced(namespace)
-          where(arel_table[:key].matches("#{namespace}:%"))
+          case ActiveRecord::Base.connection.adapter_name
+          when 'PostgreSQL'
```

Comment:
**suggestion:** In most cases, `ActiveRecord::Base.connection.adapter_name` will return `'PostgreSQL'`. But there are cases where this might no be true, even if PostgreSQL is used.

There are so much factors, when this value could look different e.g. wrapper for database driver, different databases driver or even database adapter naming in `database.yml` could alter `AR.adapter_name` due to unspoken conventions of ActiveRecord. 

And in general, even huge PG fans still can't agree on how to capitalize PostgreSQL name.

I would recommend to go with this as a method to detect PG adapters.
`ActiveRecord::Base.connection.adapter_name =~ /postg/`